File: app/src/main/java/com/ivanmagda/habito/models/ResetFrequency.java
package com.ivanmagda.habito.models;

public class ResetFrequency {

    public enum Type {
        DAY(0), WEEK(1), MONTH(2), YEAR(3), NEVER(4);

        private final int value;

        Type(int value) {
            this.value = value;
        }

        public int getValue() {
            return value;
        }
    }

    public static final String DAY = "Day";
    public static final String WEEK = "Week";
    public static final String MONTH = "Month";
    public static final String YEAR = "Year";
    public static final String NEVER = "Never";

    public static final String[] ALL = new String[]{DAY, WEEK, MONTH, YEAR, NEVER};

    private Type mType;

    public ResetFrequency() {
        this.mType = Type.NEVER;
    }

    public ResetFrequency(Type type) {
        this.mType = type;
    }

    public ResetFrequency(String stringType) {
        this.mType = typeFrom(stringType);
    }

    public Type getType() {
        return mType;
    }

    public String getTypeName() {
        return ResetFrequency.stringFor(mType);
    }

    public void setType(Type type) {
        this.mType = type;
    }

    public void setType(String typeString) {
        this.mType = ResetFrequency.typeFrom(typeString);
    }

    public static String stringFor(Type type) {
        switch (type) {
            case DAY:
                return DAY;
            case WEEK:
                return WEEK;
            case MONTH:
                return MONTH;
            case YEAR:
                return YEAR;
            case NEVER:
                return NEVER;
            default:
                throw new IllegalArgumentException("Unsupported frequency type");
        }
    }

    public static Type typeFrom(String stringType) {
        switch (stringType) {
            case DAY:
                return Type.DAY;
            case WEEK:
                return Type.WEEK;
            case MONTH:
                return Type.MONTH;
            case YEAR:
                return Type.YEAR;
            case NEVER:
                return Type.NEVER;
            default:
                throw new IllegalArgumentException("Illegal type name");
        }
    }

}

File: app/src/main/java/com/ivanmagda/habito/models/ResetFrequency.kt
package com.ivanmagda.habito.models

class ResetFrequency(type: Type = Type.NEVER) {

    var type = type
        private set(value) {
            field = value
        }

    val typeName: String
        get() = ResetFrequency.stringFor(type)

    enum class Type(val value: Int) {
        DAY(0), WEEK(1), MONTH(2), YEAR(3), NEVER(4)
    }

    constructor(stringType: String) : this() {
        type = typeFrom(stringType)
    }

    fun setType(typeString: String) {
        type = typeFrom(typeString)
    }

    companion object {

        val DAY = "Day"
        val WEEK = "Week"
        val MONTH = "Month"
        val YEAR = "Year"
        val NEVER = "Never"

        val ALL = arrayOf(DAY, WEEK, MONTH, YEAR, NEVER)

        fun stringFor(type: Type?): String {
            return when (type) {
                ResetFrequency.Type.DAY -> DAY
                ResetFrequency.Type.WEEK -> WEEK
                ResetFrequency.Type.MONTH -> MONTH
                ResetFrequency.Type.YEAR -> YEAR
                ResetFrequency.Type.NEVER -> NEVER
                else -> throw IllegalArgumentException("Unsupported frequency type")
            }
        }

        fun typeFrom(stringType: String): Type {
            return when (stringType) {
                DAY -> Type.DAY
                WEEK -> Type.WEEK
                MONTH -> Type.MONTH
                YEAR -> Type.YEAR
                NEVER -> Type.NEVER
                else -> throw IllegalArgumentException("Illegal type name")
            }
        }
    }
}
